### PR TITLE
Workflow submission files to metadata

### PR DESCRIPTION
**Pull request by cjllanwarne**
E.g.:
screen shot 2016-08-29 at 2 56 34 pm

### CODE REVIEW

File: core/src/main/scala/cromwell/core/WorkflowMetadataKeys.scala [hunk 1]
```diff
@@ -13,4 +13,9 @@ object WorkflowMetadataKeys {
   val WorkflowLog = "workflowLog"
   val Failures = "failures"
   val WorkflowRoot = "workflowRoot"
+
+  val SubmissionSection = "submittedFiles"
```
Comment (cjllanwarne) [inline core/src/main/scala/cromwell/core/WorkflowMetadataKeys.scala]: or should I underscore: `submitted_files`?

**Comment by Horneth** [comment]
👍

File: engine/src/main/scala/cromwell/engine/workflow/workflowstore/WorkflowStoreActor.scala [hunk 2]
```diff
@@ -144,11 +145,15 @@ case class WorkflowStoreActor(store: WorkflowStore, serviceRegistryActor: ActorR
...
     */
-  private def registerIdWithMetadataService(id: WorkflowId): Unit = {
+  private def registerSubmissionWithMetadataService(id: WorkflowId, sourceFiles: WorkflowSourceFiles): Unit = {
     val submissionEvents = List(
       MetadataEvent(MetadataKey(id, None, WorkflowMetadataKeys.SubmissionTime), MetadataValue(OffsetDateTime.now.toString)),
       MetadataEvent.empty(MetadataKey(id, None, WorkflowMetadataKeys.Inputs)),
-      MetadataEvent.empty(MetadataKey(id, None, WorkflowMetadataKeys.Outputs))
+      MetadataEvent.empty(MetadataKey(id, None, WorkflowMetadataKeys.Outputs)),
+
+      MetadataEvent(MetadataKey(id, None, WorkflowMetadataKeys.SubmissionSection, WorkflowMetadataKeys.SubmissionSection_Workflow), MetadataValue(sourceFiles.wdlSource)),
```
Comment (geoffjentry) [inline engine/src/main/scala/cromwell/engine/workflow/workflowstore/WorkflowStoreActor.scala]: Considering how similar these all are, having some function representing the common parts would cut down on the visual chatter - although I can't say that I really care all that much here

**Comment by geoffjentry** [comment]
👍

**Comment by kshakir** [comment]
Do these database rows need to be migrated from 0.19->0.21 also? Is this already being done by another PR?

**Pull request merged**